src/main/webapp/app/admin/user-management/user-management.component.ts

```diff
@@ -100,14 +105,20 @@ export class UserManagementComponent implements OnInit, OnDestroy {
      * Retrieve the list of users from the user service for a single page in the user management based on the page, size and sort configuration
      */
     loadAll() {
+        this.loading = true;
         this.userService
```

Comment:
I think we are fine for now.

Reply:
Another problem I see in this is, that with http you don't have a guarantee in which order the separate requests arrive - that could possibly lead to a wrong search result if one request is delayed for some networking reason

Reply:
I defenetly see your point, but to be honest I think this is really an edge case and not critical. If I understood it correctly, we are supposed to focus on functionality during review. So in my opinion this is fine, as well.

Reply:
I agree to @TobiasPr that we should optimize this at least a bit. I think a key based search is too much load on the server. I would prefer if the search is triggered when the user changes the focus of the field and when the user does not type for at least 3s (I guess this is what you refer to as debounce time).

Also the search should only be triggered if the user typed in at least 3 characters, otherwise you automatically get too many search results

Reply:
All right. Then I will implement that.

Reply:
I implemented, that the search only starts when the input field loses focus. It also has a debounce time of 1 sec. I tried it with 3 sec but thats way to much. In my opinion I would also remove 1 sec of debounce time, since you already have a delay since the user must actively leave the focus of the input. This ensures that you don't have too much traffic on the server.

I implemented that the search works when the user inserted at least 3 character or none.